file/random_access_file_reader.cc

```diff
@@ -41,6 +41,14 @@ IOStatus RandomAccessFileReader::Read(const IOOptions& opts, uint64_t offset,
   (void)aligned_buf;
 
   TEST_SYNC_POINT_CALLBACK("RandomAccessFileReader::Read", nullptr);
+
+  // A paranoid check: modify scratch a little bit, so in case underlying
+  // FileSystem doesn't fill the buffer but return succee and `scratch` returns
+  // contains a previous block, returned value will not pass checksum.
+  if (n > 0 && scratch != nullptr) {
+    scratch[0]++;
```

Comment:
We don't use scratch for direct IO I think. It doesn't apply to Zippy, so should suffice for now. But longer term, we'd need to poison the aligned buffer for direct IO.